packages/leap/lib/src/leap_game.dart

```diff
@@ -46,10 +66,23 @@ class LeapGame extends FlameGame with HasTrackedComponents {
     AssetBundle? bundle,
     Images? images,
     Map<String, TiledObjectHandler> tiledObjectHandlers = const {},
+    LeapMapTransition Function(LeapGame) mapTransitionFactory =
+        LeapMapTransition.defaultFactory,
```

Comment:
I think the factory isn't necessary, you could just pass in the Transition component directly. Unless there is something I'm missing?

Reply:
I believe having a factory is worth because the developer might now want to use the same component instance for all of their transitions, or even maybe the it will always be the same component but the information on LeapGame will influence how that component is instantiated, etc, so I feel we should keep the factory so this is more flexible. Does it makes sense?

Reply:
Hm yes, but is it necessary to make that be the parameter? Meaning, does it have to be a `Function` to a transition instead of a `Transition` itself to make that possible. If this is always invoked directly, the client code could decide to either pass in a raw constructed `Transition` or invoke a factory/function themselves when calling this function.

In theory, we'd only need to accept a factory/function here if the Map itself is going to call the factory multiple times

Reply:
not super relevant, but: late binding is also a valid reason to pass in a function to an instance instead of an instance